utils/common/src/main/java/org/apache/brooklyn/config/ConfigInheritance.java

```diff
@@ -20,31 +20,60 @@
 
 import java.io.Serializable;
 
+import org.apache.brooklyn.util.text.Strings;
+
 import com.google.common.annotations.Beta;
 
 @SuppressWarnings("serial")
 public abstract class ConfigInheritance implements Serializable {
 
-    public static final ConfigInheritance ALWAYS = new Always();
+    public enum InheritanceMode {
```

Comment:
I don't like the use of `ConfigInheritance` and `InheritanceMode`. It feels like we should just have one thing, particularly given the existing implementations of `ConfigInheritance` are so simple. But `ConfigInheritance` is not `@Beta` so I'm being conservative about changing that.

Any suggestions for a cleaner way?